### PR TITLE
OLP-1426: Saved Witnesses as separate section in genesis file

### PR DESCRIPTION
**Pull request by KevinOneLedger**
1. Updated unit tests
2. Refactored function in witness store
3. Dumped the Witnesses separately to the genesis
4. Tested by starting network, 
  - deploying smart contract
  - stopping
  - saving state
  - starting back up
  - performed lock and redeem

### CODE REVIEW

File: cmd/olfullnode/devnet.go [hunk 1]
```diff
@@ -667,7 +668,14 @@ func initialState(args *testnetConfig, nodeList []node, option ethchain.ChainDri
 			Name:             node.validator.Name,
 			Amount:           *balance.NewAmountFromInt(node.validator.Power),
 		}
+		wt := identity.Witness{
+			Address:     node.validator.Address.Bytes(),
+			PubKey:      pubkey,
```
Comment (kingpinXD) [inline cmd/olfullnode/devnet.go:673]: Does a witness need this ? or just the ECDSA key is enough ?

Comment (KevinOneLedger) [reply cmd/olfullnode/devnet.go:673]: I'm not sure if the other key is really used, I just wanted to do the refactor to use an actual witness struct. The implementation of the Witness itself is something I didn't want to modify.

Comment (kingpinXD) [reply cmd/olfullnode/devnet.go:673]: makes sense .

**Review by kingpinXD — DISMISSED** [review]
lgtm

**Review by freddyli7 — APPROVED** [review]

**Pull request merged**